lib/internal/util/comparisons.js

```diff
@@ -451,10 +526,7 @@ function mapHasEqualEntry(set, map, key1, item1, strict, memo) {
 }
 
 function mapEquiv(a, b, strict, memo) {
-  if (a.size !== b.size)
```

Comment:
Why did you inline the size checks rather than have them in mapEquiv (and setEquiv)?

Reply:
I moved them to the beginning of the checks. Now, sets will fail early in case the size does not match. Earlier they had significantly more work for to reach this check.